### PR TITLE
configurable run test verbosity

### PR DESCRIPTION
**Pull request by ipa-fxm**
follow-up of https://github.com/ros-industrial/industrial_ci/pull/255#issuecomment-357524556

@ipa-mdl 
I'm not sure how the environment variable handling works in industrial_ci
also, I was not sure whether/how to set `VERBOSE_TESTS` to `true` by default

please guide me here - or take over the branch :wink:

### CODE REVIEW

**Comment by ipa-fxm** [comment]
this feature reduces our log length from >10000 lines down to 2500!

File: industrial_ci/src/tests/source_tests.sh [hunk 1]
```diff
@@ -25,11 +25,18 @@
 ici_require_run_in_docker # this script must be run in docker
 
 #Define some verbose env vars
+#verbose build tests
```
Comment (ipa-mdl) [inline industrial_ci/src/tests/source_tests.sh]: It's just "verbose build", not just tests

**Comment by ipa-mdl** [comment]
Just some minor glitch, the rest LGTM

File: industrial_ci/src/tests/source_tests.sh (hunk 1 above)
Comment (ipa-fxm) [reply industrial_ci/src/tests/source_tests.sh]: done

File: industrial_ci/src/tests/source_tests.sh [hunk 2]
```diff
@@ -187,7 +194,7 @@ if [ "$NOT_TEST_BUILD" != "true" ]; then
 
     ici_time_start catkin_run_tests
     if [ "$BUILDER" == catkin ]; then
-        catkin run_tests --no-deps --no-status $PKGS_DOWNSTREAM $CATKIN_PARALLEL_TEST_JOBS --make-args $ROS_PARALLEL_TEST_JOBS --
+        catkin build --no-deps --catkin-make-args run_tests -- $OPT_RUN_VI --no-status $PKGS_DOWNSTREAM $CATKIN_PARALLEL_TEST_JOBS --make-args $ROS_PARALLEL_TEST_JOBS --
```
[the remark this answers is not in the document]
Comment (ipa-mdl) [reply industrial_ci/src/tests/source_tests.sh]: I am not sure, if it add something interesting.
You can just have a look and decide yourself.

**Comment by ipa-mdl** [comment]
Just had a look at this https://travis-ci.org/ros-industrial/industrial_ci/jobs/330252621
(compared to https://travis-ci.org/ros-controls/ros_control/jobs/327897154)
The output verbosity was reduced dramatically, but now the ROS_ERROR line don't have an context.

I think ` OPT_RUN_VI="-vi"` needs to be changed to ` OPT_RUN_VI="-v"` for verbose runs.
And perhaps we should make it the default.
@130s: What do you think?

**Comment by ipa-fxm** [comment]
@ipa-mdl @130s 
in bea3aa3097698fc8de54bfc817ea8fb90542dd43, I removed the `interleave-output`
Let me know what you like better

For making it default, I would need a pointer to an example showing how this is done!

File: industrial_ci/src/tests/source_tests.sh [hunk 3]
```diff
@@ -25,11 +25,18 @@
...
     OPT_VI="-vi"
 else
     OPT_VI=""
 fi
+#verbose run tests
+if [ "$VERBOSE_TESTS" ] && [ "$VERBOSE_TESTS" == true ]; then
+    OPT_RUN_V="-v"
+else
+    OPT_RUN_V=""
+fi
```
Comment (ipa-mdl) [inline industrial_ci/src/tests/source_tests.sh]:
```
if [ "$VERBOSE_TESTS" = false ]; then
    OPT_RUN_V=""
else
    OPT_RUN_V="-v"
fi
```

Comment (ipa-fxm) [reply industrial_ci/src/tests/source_tests.sh]: 😉 I thought it'd be more complicated ^^
(but I'll wait until the decision with/without `interleave` is made)

**Comment by ipa-mdl** [comment]
> in bea3aa3, I removed the interleave-output
Let me know what you like better

Yes, his was the default before and definately makes sense für the tests.
In interleaved mode you might see a lot of error line of tests that check if they occur without any chance to link them to specific tests cases.

**Comment by ipa-mdl** [comment]
I am happy :)

@130s: Should tests be verbose by default or not?

File: doc/index.rst [hunk 4]
```diff
@@ -194,6 +194,7 @@ Note that some of these currently tied only to a single option, but we still lea
 * `USE_DEB` (*DEPRECATED*: use `UPSTREAM_WORKSPACE` instead. default: true): if `true`, `UPSTREAM_WORKSPACE` will be set as `debian`. if `false`, `file` will be set. See `UPSTREAM_WORKSPACE` section for more info.
 * `USE_MOCKUP` (default: not set): reletive path to mockup packages to be used for the tests
 * `VERBOSE_OUTPUT` (default: not set): If `true`, build tool (e.g. Catkin) output prints in verbose mode.
```
Comment (130s) [inline doc/index.rst]: Now that we have `VERBOSE_TESTS` one line below, we need to explain this one better. Maybe add `except tests steps` at the end of the sentence?

Comment (130s) [reply doc/index.rst]: Now that we have `VERBOSE_TESTS` one line below, we need to explain this one better. Maybe add something like "except tests steps (see also `VERBOSE_TESTS`)" at the end of the sentence?

**Comment by ipa-fxm** [comment]
@130s @ipa-mdl 
although, I can't see @130s comments here in the issue, I got them per git notification
thus, I considered them and re-ordered/squashed the commits
now everything should be good to merge

**Pull request merged**